models/autotetraploid.slim

```diff
@@ -0,0 +1,130 @@
+	away in an individual in p2; the individuals in p2 don't do anything on their own. To identify them,
+	each individual in p1 has a "tag" that records the index of their pair in p2, so that
+	  p2.individuals[ind.tag].genomes
+	gets you the other two genomes for individual "ind" in p1. After an individual dies, we clean up their
+	paired individual by setting their fitness to 0.0 in the following generation. So, there's always more
+	individuals in p2 than in p1; the difference is individuals in p2 whose pair in p1 has just died.
+	
+	(b) For crossing-over, we pick an ordering of the mates' four genomes, then do crossing-over by picking
+	a Poisson number of uniform recombination points on each of the four pairs, with an extra crossing-over
+	at 0.0 to randomize which genome we start with. If we want non-uniform recombination, we'd have to get
+	these recombinations some other way.
+	
+	(c) Every new mutation has a property, "h", that is a vector of length 5, always starting with 0 and ending
+	with 1, that has randomly chosen dominance coefficients. Then, the contribution to fitness of having
+	k copies of a mutation is h[k] * s, where recall that: indexing is zero-based; fitness is multiplicative; and
+	since this is a nonWF model, "fitness" means probability of survival until the next generation.
+
+*/
+
+initialize() {
+	initializeSLiMModelType("nonWF");
+		
+	defineConstant("K", 100);
+	defineConstant("L", 1e6); // total bp length
+
+	
+	initializeMutationType("m1", 0.5, "e", 0.01);
+	m1.convertToSubstitution = T;
+	
+	initializeGenomicElementType("g1", m1, 1.0);
+	initializeGenomicElement(g1, 0, L - 1);
+	initializeMutationRate(1e-7);
+	initializeRecombinationRate(1e-8);
+	defineConstant("totalLength", L * 1e-8); // total genetic length
+}
+
+reproduction(p1) {
+	mate = p1.sampleIndividuals(1);
+	ma_chroms = c(individual.genomes, p2.individuals[individual.tag].genomes);
+	pa_chroms = c(mate.genomes, p2.individuals[mate.tag].genomes);
+	pairing = sample(seqLen(4), 4);
+	// including 0.0 starts with the second chromosome instead of the first
+	breakpoints1 = asInteger(c(runif(1) < 0.5 ? 0 else NULL, floor(L * runif(rpois(1, totalLength)))));
+	breakpoints2 = asInteger(c(runif(1) < 0.5 ? 0 else NULL, floor(L * runif(rpois(1, totalLength)))));
+	offspring = p1.addRecombinant(ma_chroms[0], pa_chroms[pairing[0]], breakpoints1,
+											ma_chroms[1], pa_chroms[pairing[1]], breakpoints2);
+	breakpoints1 = asInteger(c(runif(1) < 0.5 ? 0 else NULL, floor(L * runif(rpois(1, totalLength)))));
+	breakpoints2 = asInteger(c(runif(1) < 0.5 ? 0 else NULL, floor(L * runif(rpois(1, totalLength)))));
+	other = p2.addRecombinant(ma_chroms[2], pa_chroms[pairing[2]], breakpoints1,
+											ma_chroms[3], pa_chroms[pairing[3]], breakpoints2);
+	tag = runif(1);
+	offspring.tagF = tag;
+	other.tagF = tag;
+}
+
+mutation(m1) {
+	// the fitness effect of having k alleles will be h[k] * s
+	mut.setValue("h", c(0.0, sort(runif(3)), 1.0));
```

Comment:
The "sort" is optional, I guess, and assumes some kind of linear dosage-response thing going on; one could certainly imagine other relationships, right?  But this seems a reasonable default.

Reply:
yep